photutils/psf/psfphotometry.py

```diff
@@ -0,0 +1,282 @@
+        
+        Parameters
+        ----------
+        fit_model : Fittable2DModel
+        star_group : ~astropy.table.Table
+        
+        Returns
+        -------
+        param_tab : ~astropy.table.Table
+            Table that contains the fitted parameters.
+        """
+
+        param_tab = Table([[], [], [], [], []],
+                          names=('id', 'group_id', 'x_fit', 'y_fit',
+                                 'flux_fit'),
+                          dtype=('i4', 'i4', 'f8', 'f8', 'f8'))
+        if np.size(fit_model) == 1:
+            param_tab.add_row([[star_group['id'][0]],
+                               [star_group['group_id'][0]],
+                               [getattr(fit_model,'x_0').value],
+                               [getattr(fit_model, 'y_0').value],
+                               [getattr(fit_model, 'flux').value]])
+        else:
+            for i in range(np.size(fit_model)):
+                param_tab.add_row([[star_group['id'][i]],
+                                   [star_group['group_id'][i]],
+                                   [getattr(fit_model,'x_0_'+str(i)).value],
+                                   [getattr(fit_model, 'y_0_'+str(i)).value],
+                                   [getattr(fit_model, 'flux_'+str(i)).value]])
+        return param_tab
+
+    def _get_shape_and_data(self, shape, star_group, image):
+        """
+        Parameters
+        ----------
+        shape : tuple
+            Shape of a rectangular region around the center of an isolated source.
+        star_group : `astropy.table.Table`
+            Group of stars
+        image : numpy.ndarray
+
+        Returns
+        -------
+        x, y : numpy.mgrid
+            All coordinate pairs (x,y) in a rectangular region which encloses all
+            sources of the given group
+        image : numpy.ndarray
+            Pixel value
+        """
+
+        xmin = int(np.around(np.min(star_group['x_0'])) - shape[0])
+        xmax = int(np.around(np.max(star_group['x_0'])) + shape[0])
+        ymin = int(np.around(np.min(star_group['y_0'])) - shape[1])
+        ymax = int(np.around(np.max(star_group['y_0'])) + shape[1])
+        y, x = np.mgrid[ymin:ymax+1, xmin:xmax+1]
+
+        return x, y, image[ymin:ymax+1, xmin:xmax+1]
+
+    def _get_sum_psf_model(self, star_group):
+        """
+        Construct a joint psf model which consists in a sum of `self.psf`
+        whose parameters are given in `star_group`.
+
+        Parameters
+        ----------
+        star_group : `~astropy.table.Table`
+            Table from which the compound PSF will be constructed.
+            It must have columns named as `x_0`, `y_0`, and `flux_0`.
+        
+        Returns
+        -------
+        sum_psf : CompoundModel
+            `CompoundModel` instance which is a sum of the given PSF
+            models.
+        """
+
+        psf_class = type(self.psf)
+        sum_psf = psf_class(sigma=self.psf.sigma.value,
```

Comment:
How would one fix certain parameters and make sure they stay fixed for all fits?
Commonly, one would fit the sigma only on the brightest stars and then freeze it in the fit. That has two advantages: 1) The number of fit parameters is lower, makes for a faster fit and 2) sigma really is a property of the telescope and the weather at the time of observations, so it should be the same for all stars. Fits often go bad when there are several faint stars on a background, e.g. one of them gets fit with sigma = 1e5, which is essentially a constant background.
Similarly, one might want to keep x_0 and y_0 fixed. For example, imagine that I already know the positions of the stars from the HIPPARCOS catalog, which is a lot more precise than any ground-based data that I have. In that case, I only want to fit the fluxes.